jenkins/provider.go

```diff
@@ -26,13 +26,13 @@ func Provider() *schema.Provider {
 			},
 			"username": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true, // Needs to be optional to be able to run terraform validate in CICD pipelines.
 				DefaultFunc: schema.EnvDefaultFunc("JENKINS_USERNAME", nil),
 				Description: "Username to authenticate to Jenkins.",
 			},
 			"password": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true, // Needs to be optional to be able to run terraform validate in CICD pipelines.
```

Comment:
I'm not sure we should call out "in CICD pipelines" explicitly here as `terraform validate` may be used in multiple different contexts.

Reply:
https://github.com/taiidani/terraform-provider-jenkins/pull/124/commits/4e8e7cef683b4294424ee09700469851b176108c